export_snowflake/__init__.py

```diff
@@ -522,26 +528,126 @@ def flush_records(stream: str,
     db_sync.delete_from_stage(stream, s3_key)
 
 
+def transfer_data_in_parallel(s3_csv_lists, config, o, file_format_type):
+    parallel_transfer_start_time = time.time()
+    
+    # make the decision of how many processes to use later
+    pool = multiprocessing.Pool(4, initialize_s3_client_db_sync(config, o, file_format_type))
+    pool.map(download_file, s3_csv_lists)
+    pool.close()
+    pool.join()
```

Comment:
please make sure error in one process/thread causes all other threads to exit as well, else other threads will continue to do its work, and only at the end it knows about the error. Need to implement error/exception handling such way that target snowflake communicates the error back to app, but also exit all other threads immediately.

Reply:
Updated with an error queue to listen to the error signal. Once error code is detected, will populate it between the threads. Error is caught and sent back to main app.